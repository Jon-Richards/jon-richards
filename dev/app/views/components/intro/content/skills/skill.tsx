import * as React from 'react';
const CSS = require('./skills.scss');

type SkillProps = {
  /** UUID for this particular skill. */
  uuid: string;
  /** The label as it will display to the user. */
  displayLabel: string;
  /** If this is a core skill. */
  isCore: boolean;
};

/**
 * Renders a single skill.
 *
 * @example
 * <Skill
 *      uuid="abc-123"
 *      displayLabel="foo"
 *      value="bar"
 *      isActive={true}
 *      isCore={false}
 *      onClick={(e, value) => {
 *          console.log(value);
 *      }}
 * />
 */
function Skill (props: SkillProps): JSX.Element {
  const { displayLabel, isCore } = props;

  return (
    <span
      className={
        `${CSS['skill']}` +
        `${isCore ? ' ' + CSS['skill--is-core'] : ''}`
      }
    >
      {displayLabel}
    </span>
  );
}

const skillMemo = React.memo<SkillProps>(Skill);

export {skillMemo as Skill, SkillProps};