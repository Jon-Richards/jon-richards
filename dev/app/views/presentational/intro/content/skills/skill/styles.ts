import { Breakpoints } from 'Config/styles/breakpoints';
import { Colors } from 'Config/styles/colors';
import { Fonts } from 'Config/styles/fonts';

/** Styles for a single skill. */
export const STYLES = {
  skill: `
    display: none;
    margin: 0.2rem 0.2rem;
    padding: 0.4rem 1rem 0.5rem;
    width: auto;
    background: ${Colors.SWATCH_1.withLightness(15).value};
    color: ${Colors.SWATCH_2.value};
    font-family: ${Fonts.OPEN_SANS.fontFamily};
    font-weight: ${Fonts.OPEN_SANS.fontStyles.regular.weights.light};
    border-radius: 0.5rem;

    @media (min-width: ${Breakpoints[720]}) {
      display: block;
    }
  `,
  coreSkill: `
    display: none;
    margin: 0.2rem 0.2rem;
    padding: 0.4rem 1rem;
    width: auto;
    background: ${Colors.SWATCH_1.value};
    color: ${Colors.SWATCH_2.withLightness(30).value};

    @media (min-width: ${Breakpoints[480]}) {
      display: flex;
    }
  `
};