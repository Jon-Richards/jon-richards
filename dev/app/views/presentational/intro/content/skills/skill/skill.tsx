/** @jsx jsx */

import * as React from 'react';
import { STYLES } from './styles';
import { css, jsx } from '@emotion/core';

type SkillProps = {
  /** UUID for this particular skill. */
  uuid: string;
  /** The label as it will display to the user. */
  displayLabel: string;
  /** If this is a core skill. */
  isCore: boolean;
};

/**
 * Renders a single skill.
 *
 * @example
 * <Skill
 *      uuid="abc-123"
 *      displayLabel="foo"
 *      value="bar"
 *      isActive={true}
 *      isCore={false}
 *      onClick={(e, value) => {
 *          console.log(value);
 *      }}
 * />
 */
function Skill (props: SkillProps): JSX.Element {
  const { displayLabel, isCore } = props;

  return (
    <span
      css={css(isCore ? STYLES.skill : STYLES.coreSkill)}
      data-testid="intro__skill"
    >
      {displayLabel}
    </span>
  );
}

const skillMemo = React.memo<SkillProps>(Skill);

export {skillMemo as Skill, SkillProps};