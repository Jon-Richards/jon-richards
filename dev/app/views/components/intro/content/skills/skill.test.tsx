import * as React from 'react';
import {shallow, mount, render} from 'enzyme';
import {Skill, SkillProps} from './skill';
import { v4 as uuid } from 'uuid';

describe('The Skill component.', () => {
  it('Should render without throwing an error.', () => {
    expect(
      shallow(
        <Skill
          uuid={uuid()}
          displayLabel="Foo"
          isCore={true}
        />
      )
      .contains('Foo')
    ).toBe(true);

    expect(
      shallow(
        <Skill
          uuid={uuid()}
          displayLabel="Bar"
          isCore={true}
        />
      )
      .contains('Foo')
    ).toBe(false);
  });
});