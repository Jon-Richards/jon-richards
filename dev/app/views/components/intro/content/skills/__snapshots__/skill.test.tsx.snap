// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Skill component. Should match the snapshot. 1`] = `
<span
  className="skill skill--is-core"
>
  Bar
</span>
`;
